OpenRA.Mods.Cnc/Traits/SupportPowers/GrantPrerequisiteChargeDrainPower.cs

```diff
@@ -183,17 +183,15 @@ public override void Activate(Order order)
 
 			public override string IconOverlayTextOverride()
 			{
-				var info = Info as GrantPrerequisiteChargeDrainPowerInfo;
-				if (info == null || !Active)
+				if (!(Info is GrantPrerequisiteChargeDrainPowerInfo info) || !Active)
```

Comment:
It would take some intentional hackery gymnastics for anyone to create a situation where `Info` isn't `GrantPrerequisiteChargeDrainPowerInfo`, so IMO we should just drop the check and go with the clearer:
```csharp
if (!Active)
	return null;

var info = (GrantPrerequisiteChargeDrainPowerInfo)Info;
return active ? info.ActiveText : available ? info.AvailableText : null;
```